lib/Optimizer/Transforms/RegToMem.cpp

```diff
@@ -287,36 +461,32 @@ class RegToMemPass : public cudaq::opt::impl::RegToMemBase<RegToMemPass> {
     // equivalence classes.
     RegToMemAnalysis analysis(func);
     if (analysis.failed()) {
-      LLVM_DEBUG(llvm::dbgs() << func << "\n");
-      func.emitOpError("register analysis failed\n");
-      signalPassFailure();
+      LLVM_DEBUG(llvm::dbgs()
+                 << "regtomem dataflow analysis for \"" << func.getName()
+                 << "\" failed.\nAll wire uses must remain disjoint from the "
+                    "source null/borrow to the sink/return operation.\n");
       return;
     }
 
     // 2) Replace each quake.null_wire with a quake.alloca.
     SmallVector<Value> allocas(analysis.getCardinality());
-    for (auto nwire : analysis.getNullWires()) {
-      OpBuilder builder(nwire);
+    SmallVector<Value> borrowAllocas;
+    for (auto *nwire : analysis.getWires()) {
+      OpBuilder builder(ctx);
+      const bool fromWire = isa<quake::BorrowWireOp>(nwire);
+      if (fromWire)
+        builder.setInsertionPointToStart(&func.getBody().front());
```

Comment:
Won't this place the `quake.alloca`s coming from borrow wires in reverse order? If so, I would expect some of the existing tests to fail (e.g. `targettests/execution/mapping_test-1.cpp`).